source/fuzz/transformation_record_synonymous_constants.cpp

```diff
@@ -19,8 +19,61 @@ namespace spvtools {
 namespace fuzz {
 
 namespace {
-bool IsScalarZeroConstant(const opt::analysis::Constant* constant) {
-  return constant->AsScalarConstant() && constant->IsZero();
+
+// Returns true if the two given constants are equivalent
+// (the description of IsApplicable specifies the conditions they must satisfy
+// to be considered equivalent)
+bool AreEquivalentConstants(opt::IRContext* ir_context,
+                            const opt::analysis::Constant& constant1,
+                            const opt::analysis::Constant& constant2) {
+  // Check that the type ids are the same
+  // TODO(https://github.com/KhronosGroup/SPIRV-Tools/issues/3536): Somehow
+  // relax this for integers (so that unsigned integer and signed integer are
+  // considered the same type)
+  uint32_t type_id1 = ir_context->get_type_mgr()->GetId(constant1.type());
+  uint32_t type_id2 = ir_context->get_type_mgr()->GetId(constant2.type());
```

Comment:
Change this so that it uses the def-use manager to get the instruction for each constant id, and then the type from that instruction.

Reply:
Done